### PR TITLE
DOCS: Document functions that return lists

### PR DESCRIPTION
**Pull request by tlimoncelli**
Fixes https://github.com/StackExchange/dnscontrol/issues/2339

### CODE REVIEW

File: documentation/examples.md [hunk 1]
```diff
@@ -123,6 +123,18 @@ D("example.tld", REG, DSP, ...
...
+## Set default records modifiers ##
+{% code title="dnsconfig.js" %}
+```javascript
+DEFAULTS(
+    NAMESERVER_TTL('24h'),
+    DefaultTTL('12h'),
+    CF_PROXY_DEFAULT_OFF
+);
+```
+
```
Comment (hmoffatt) [inline documentation/examples.md:139]: There's an `{% endcode %}` missing here.

Comment (tlimoncelli) [reply documentation/examples.md:139]: Ooops! Thanks for catching that!

**Comment by tlimoncelli** [comment]
Fixing endcode issue: https://github.com/StackExchange/dnscontrol/pull/2360

File: documentation/examples.md [hunk 2]
```diff
@@ -21,35 +21,37 @@ D('example.com', REG, DnsProvider('GCLOUD'),
...
+D("example.com", REG, DnsProvider(DSP_R53),
+    A("@", addrA), // 1.2.3.4
+    A("www", addrA + 1), // 1.2.3.5
 )
 ```
 {% endcode %}
 
 {% hint style="info" %}
-**NOTE**: The [`IP()`](functions/global/IP.md) function doesn't currently support IPv6 (PRs welcome!).  IPv6 addresses are strings.
+**NOTE**: The [`IP()`](functions/global/IP.md) function doesn"t currently support IPv6 (PRs welcome!).  IPv6 addresses are strings.
```
Comment (cafferata) [inline documentation/examples.md:54]: @tlimoncelli this single quote should not be replaced.

File: documentation/examples.md [hunk 3]
```diff
@@ -190,6 +199,11 @@ We can then use the macros as such:
...
     FASTMAIL_MX,
-    FASTMAIL_DKIM('example.com')
+    FASTMAIL_DKIM("example.com")
 )
 ```
+{% endcode %}
+
+### More advanced examples
+
+See the [Code Tricks](advanced-features/code-tricks) page.
```
Comment (cafferata) [inline documentation/examples.md:209]: @tlimoncelli this link is incorrect, this should be `code-tricks.md`.

```diff
-See the [Code Tricks](advanced-features/code-tricks) page.
+See the [Code Tricks](code-tricks.md) page.
```

_https://docs.dnscontrol.org/getting-started/examples#more-advanced-examples_

**Pull request merged**